### ISSUE TITLE
OpenBLAS 0.3.8 fails to build with DYNAMIC_ARCH=1 on s390x architecture

### ISSUE THREAD

**Issue by susilehtola** [xianyi/OpenBLAS#2406]
OpenBLAS 0.3.8 fails to build with DYNAMIC_ARCH=1 on s390x. After addressing the typo in `dynamic_zarch.c` fixed by #2405, linking fails with
```
/lib/gcc/s390x-redhat-linux/10/../../..  -lc  
/usr/bin/ld: ../libRblas-r0.3.8.a(dynamic_zarch.o): in function `gotoblas_corename':
/builddir/build/BUILD/openblas-0.3.8/Rblas/driver/others/dynamic_zarch.c:30: undefined reference to `gotoblas_Z15'
/usr/bin/ld: ../libRblas-r0.3.8.a(dynamic_zarch.o): in function `force_coretype':
/builddir/build/BUILD/openblas-0.3.8/Rblas/driver/others/dynamic_zarch.c:81: undefined reference to `gotoblas_Z15'
/usr/bin/ld: ../libRblas-r0.3.8.a(dynamic_zarch.o): in function `gotoblas_corename':
/builddir/build/BUILD/openblas-0.3.8/Rblas/driver/others/dynamic_zarch.c:30: undefined reference to `gotoblas_Z15'
/usr/bin/ld: ../libRblas-r0.3.8.a(dynamic_zarch.o): in function `force_coretype':
/builddir/build/BUILD/openblas-0.3.8/Rblas/driver/others/dynamic_zarch.c:81: undefined reference to `gotoblas_Z15'
```

**Comment by susilehtola** [comment]
Looks like there is no Z15 kernel, proposed fix #2407 .

**Comment by martin-frbg** [comment]
Correct, thanks.
